opencog/atomspace/Handle.h

```diff
@@ -71,13 +73,33 @@ friend class AtomspaceHTabler;    // persistance
     AtomPtr resolve_ptr();
     static void set_resolver(const AtomTable*);
     static void clear_resolver(const AtomTable*);
+
+    // Set the UUID in a handle. This is the safe version. Use  this version
+    // if you have any doubts at all about the possibility that the handle
+    // may have been resolved. This sets the new UUID and also calls 
+    // resets the shared_ptr AtomPtr so it doesn't point to anything,
+    // the reference is decremented, and the object deleted if this is the
+    // last one. This is used by AtomTable to avoid having to construct
+    // temporaries while iterating when the iteration itself can resolve the
+    // AtomPtr in order to clear the AtomPtr reference so it won't be pointing
+    // to an atom that has a UUID that is different than the UUID passed
+    // to set_uuid.    
+    void set_uuid(const UUID uuid) { _ptr.reset(); _uuid = uuid; }
```

Comment:
I don't like this, this is going backwards to the bad old days, where everything was done with UUID's. The UUID is really not meant to be used this way; it is intended only to guaranteed that atoms in different atomspaces (i.e. on different network servers) have a unique identifier.  They aren't supposed to be used for referencing that are within the same address space (i.e within the same process, or shared by different threads.